Item 7002: Refactor domain designer components to share more with base components

cnathe opened the pull request:
#### Rationale
Changes have been made to consolidate some of the shared parts of the various domain designer components in the @labkey/components package. This PR pulls in those updates via a version update to that package.

Also, there were some simplifications that were made for the experiment module wrapper of the SampleTypeDesigner.tsx component to make it more consistent with other domain designer regarding URL param validation and error handling.

#### Related Pull Requests
* https://github.com/LabKey/labkey-ui-components/pull/195
* https://github.com/LabKey/sampleManagement/pull/218
* https://github.com/LabKey/testAutomation/pull/280

#### Changes
* update @labkey/components package version in relevant platform module package.json files
* SampleTypeDesigner.tsx component refactor for consistency with other domain designers
* Remove sample set special handling for reserved field names on create sample set (instead throw error like other domain designers)

cnathe committed:
Item 7002: Update to latest @labkey/compoments package with domain designer refactors

cnathe committed:
Item 7002: Update to latest @labkey/compoments alpha package version

cnathe committed:
Update to latest @labkey/components alpha package version

cnathe committed:
Merge branch 'develop' into fb_smSampleTypeDesignerPolish

# Conflicts:
#	assay/package-lock.json
#	assay/package.json
#	core/package-lock.json
#	core/package.json
#	experiment/package-lock.json
#	experiment/package.json
#	list/package-lock.json
#	list/package.json
#	query/package-lock.json
#	query/package.json

cnathe committed:
Update to latest @labkey/components alpha package version

cnathe committed:
Update to latest @labkey/components alpha package version

cnathe committed:
Update to latest @labkey/components alpha package version

cnathe committed:
Update to latest @labkey/components alpha package version

cnathe committed:
Merge branch 'develop' into fb_smSampleTypeDesignerPolish

# Conflicts:
#	assay/package-lock.json
#	assay/package.json
#	core/package-lock.json
#	core/package.json
#	experiment/package-lock.json
#	experiment/package.json
#	list/package-lock.json
#	list/package.json
#	query/package-lock.json
#	query/package.json

cnathe committed:
Update to latest @labkey/components alpha package

cnathe committed:
Update to latest @labkey/components alpha package version

cnathe committed:
Update to latest @labkey/components alpha package version

cnathe committed:
Item 7002: SampleTypeDesigner.tsx updates for error message and URL param handling

cnathe committed:
Update to latest @labkey/components alpha package version

labkey-susanh commented on experiment/src/client/SampleTypeDesigner/SampleTypeDesigner.tsx:
```diff
@@ -14,73 +14,65 @@
-            List<IBannerMessage>();
+        let message;
+        if (action === UPDATE_SAMPLE_SET_ACTION && !this.getRowIdParam()) {
+            message = 'RowId parameter not supplied, unable to determine which Sample Type to edit.';
```
```suggestion
            message = 'RowId parameter not supplied. Unable to determine which Sample Type to edit.';
```
Are we using Sample Type in LKS now instead of Sample Set?

cnathe replied:
Nope, good catch. I'll change the usages / text in this component back to "Sample Set".

cnathe committed:
Update to latest @labkey/components alpha package version

labkey-susanh commented on experiment/src/client/SampleTypeDesigner/SampleTypeDesigner.tsx:
```diff
@@ -92,129 +84,88 @@ export class App extends React.PureComponent<any, Partial<IAppState>> {
-                    this.setState(() => ({
-                        messages: messages.set(0, {message: error.exception, messageType: 'danger'})
-                    }));
+                    this.setState(() => ({message: error.exception, isLoading: false}));
```
Do we want to resolve that exception to something possibly more friendly to the user?

cnathe replied:
You mean you don't like the current error response here: "MaterialSource not found: 10852"? Yeah, I'll update that.

labkey-susanh commented on experiment/src/client/SampleTypeDesigner/SampleTypeDesigner.tsx:
```diff
@@ -92,129 +84,88 @@ export class App extends React.PureComponent<any, Partial<IAppState>> {
-                this.setState(() => ({
-                    messages: messages.set(0, {message: error.exception, messageType: 'danger'})
-                }));
+                this.setState(() => ({message: error.exception, isLoading: false}));
```
Same here about resolving that exception message.

cnathe replied:
Yes, done.

cnathe committed:
Merge branch 'develop' into fb_smSampleTypeDesignerPolish

cnathe committed:
Item 7002: Use "Sample Set" as text in the LKS page for now (instead of "Sample Type")

cnathe committed:
Update to latest @labkey/components alpha package version

cnathe committed:
Item 7002: Fix for Sample Type validation check on creation to check for matching parent alias names and field names
- note this was already being validated in the sample type update case, but was not happening on create

cnathe committed:
Item 7002: Remove sample type creation special handling for reserved fields
- remove code that was uniqifying the field name
- add a test case in SampleSetTest

cnathe committed:
Update to latest @labkey/components alpha package version

cnathe committed:
Merge branch 'develop' into fb_smSampleTypeDesignerPolish

cnathe committed:
Update to @labkey/components release version 0.41.0
